src/runtime/shiki/event-handler.ts

```diff
@@ -1,23 +1,27 @@
-import type { Lang } from 'shiki-es'
-import { eventHandler, getQuery } from 'h3'
+import type { BuiltinLanguage, BuiltinTheme } from 'shikiji'
+import { loadWasm } from 'shikiji'
+import { eventHandler, getQuery, lazyEventHandler } from 'h3'
 import { useShikiHighlighter } from './highlighter'
-import type { Theme, TokenStyleMap } from './types'
 
-export default eventHandler(async (event) => {
-  const { code, lang, theme: themeString, highlights: highlightsString } = getQuery(event)
-  const theme = JSON.parse(themeString as string)
-  const highlights = JSON.parse(highlightsString as string) as number[]
-
-  const shikiHighlighter = useShikiHighlighter({})
+export default lazyEventHandler(async () => {
```

Comment:
WDYT about removing this handler and move this login directly inside default highlighter function (in `src/runtime/parser/shiki.ts`)


/cc @Atinux

Reply:
It's a bit tricky, as importing the `.wasm` directly isn't a standard yet, but CF supports it and it's the only way you can import a wasm in CF (it does not allow to initiate from buffer, for security reasons maybe?). So I feel it would be risky and costly to do so directly in `shikiji` at this moment.